portfolio/src/main/webapp/script.js

```diff
@@ -86,3 +86,17 @@ function changePattern(pattern) {
     }
   }
 }
+
+async function retrieveComments() {
+    const response = await fetch("/data");
+    const message_json = await response.json();
+
+    var commentHtml = "";
+    var commentContainer = document.getElementById("comment-container");
+    for (var i = 0; i< message_json.length ; i++) {
```

Comment:
Nit: use a forEach loop here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach. Shorter and helps avoid off-by-one errors.

Reply:
resolved all the issues.